Flaky test somewhere in grpc/test/...

The errors before the panic are likely unrelated (invalid UTF-8 happens intentionally due to a test case).  Unfortunately, the stack trace isn't very helpful - we can't see the test that was running because it was in the service handler:

https://github.com/grpc/grpc-go/blob/030824531b6588b2f7fb381c9efa96c560693b83/test/end2end_test.go#L4623

```
E0815 18:56:42.074802    7030 clientconn.go:1282] Subchannel health check is unimplemented at server side, thus health check is disabled
E0815 18:56:58.246633    7030 http2_server.go:836] transport: failed to marshal rpc status: code:13 message:"\377\376\375" details:<type_url:"type.googleapis.com/grpc.testing.Empty" > , error: proto: field "google.rpc.Status.Message" contains invalid UTF-8
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x0 pc=0xb99cdf]
goroutine 50276 [running]:
google.golang.org/grpc/test.(*funcServer).FullDuplexCall(0xc000e6c480, 0xe74ca0, 0xc000ae8470, 0xe71d00, 0xc000e6c480)
	/home/travis/gopath/src/google.golang.org/grpc/test/end2end_test.go:4623 +0x5f
google.golang.org/grpc/test/grpc_testing._TestService_FullDuplexCall_Handler(0xce74c0, 0xc000e6c480, 0xe71c40, 0xc000a6c300, 0x137e3f8, 0xc000a5e300)
	/home/travis/gopath/src/google.golang.org/grpc/test/grpc_testing/test.pb.go:817 +0xce
google.golang.org/grpc.(*Server).processStreamingRPC(0xc0001b8160, 0xe762c0, 0xc000be9080, 0xc000a5e300, 0xc000e6c510, 0x1356ac0, 0x0, 0x0, 0x0)
	/home/travis/gopath/src/google.golang.org/grpc/server.go:1199 +0x1536
google.golang.org/grpc.(*Server).handleStream(0xc0001b8160, 0xe762c0, 0xc000be9080, 0xc000a5e300, 0x0)
	/home/travis/gopath/src/google.golang.org/grpc/server.go:1279 +0x12e6
google.golang.org/grpc.(*Server).serveStreams.func1.1(0xc000b900d0, 0xc0001b8160, 0xe762c0, 0xc000be9080, 0xc000a5e300)
	/home/travis/gopath/src/google.golang.org/grpc/server.go:710 +0xad
created by google.golang.org/grpc.(*Server).serveStreams.func1
	/home/travis/gopath/src/google.golang.org/grpc/server.go:708 +0xb9
FAIL	google.golang.org/grpc/test	279.616s
```

--- a/test/end2end_test.go
+++ b/test/end2end_test.go
@@ -7113,7 +7113,7 @@ func (s) TestGoAwayThenClose(t *testing.T) {
 	}
 	s1 := grpc.NewServer()
 	defer s1.Stop()
-	ts1 := &funcServer{
+	ts := &funcServer{
 		unaryCall: func(ctx context.Context, in *testpb.SimpleRequest) (*testpb.SimpleResponse, error) {
 			return &testpb.SimpleResponse{}, nil
 		},
@@ -7126,7 +7126,7 @@ func (s) TestGoAwayThenClose(t *testing.T) {
 			return err
 		},
 	}
-	testpb.RegisterTestServiceServer(s1, ts1)
+	testpb.RegisterTestServiceServer(s1, ts)
 	go s1.Serve(lis1)
 
 	conn2Established := grpcsync.NewEvent()
@@ -7136,17 +7136,13 @@ func (s) TestGoAwayThenClose(t *testing.T) {
 	}
 	s2 := grpc.NewServer()
 	defer s2.Stop()
-	ts2 := &funcServer{unaryCall: func(ctx context.Context, in *testpb.SimpleRequest) (*testpb.SimpleResponse, error) {
-		return &testpb.SimpleResponse{}, nil
-	}}
-	testpb.RegisterTestServiceServer(s2, ts2)
+	testpb.RegisterTestServiceServer(s2, ts)
 	go s2.Serve(lis2)
 
 	r, rcleanup := manual.GenerateAndRegisterManualResolver()
 	defer rcleanup()
 	r.InitialState(resolver.State{Addresses: []resolver.Address{
 		{Addr: lis1.Addr().String()},
-		{Addr: lis2.Addr().String()},
 	}})
 	cc, err := grpc.DialContext(ctx, r.Scheme()+":///", grpc.WithInsecure())
 	if err != nil {
@@ -7162,6 +7158,11 @@ func (s) TestGoAwayThenClose(t *testing.T) {
 	if err != nil {
 		t.Fatalf("FullDuplexCall(_) = _, %v; want _, nil", err)
 	}
+
+	r.UpdateState(resolver.State{Addresses: []resolver.Address{
+		{Addr: lis1.Addr().String()},
+		{Addr: lis2.Addr().String()},
+	}})
 
 	// Send GO_AWAY to connection 1.
 	go s1.GracefulStop()

